WiserTaskScheduler/WiserTaskScheduler/Modules/ServerMonitors/Services/ServerMonitorsService.cs

```diff
@@ -0,0 +1,433 @@
+            body = $"Drive {drive.Name} only has {percentage}% space left, this is below the threshold of {threshold}";
+
+            await logService.LogInformation(logger, LogScopes.RunStartAndStop, monitorItem.LogSettings, $"Drive {drive} has {drive.TotalFreeSpace}Bytes of free space", configurationServiceName, monitorItem.TimeId, monitorItem.Order);
+
+            //Check if the threshold is higher then the free space available.
+            if (percentage < threshold)
+            {
+                await logService.LogInformation(logger, LogScopes.RunStartAndStop, monitorItem.LogSettings, $"Drive {drive.Name} only has {percentage}% space left, this is below the threshold of {threshold}", configurationServiceName, monitorItem.TimeId, monitorItem.Order);
+
+                //Only send an email if the Drive threshold hasn't already been reached.
+                if (!emailDrivesSent[drive.Name])
+                {
+                    emailDrivesSent[drive.Name] = true;
+                    await gclCommunicationsService.SendEmailAsync(receiver, subject, body);
+                }
+            }
+            else
+            {
+                emailDrivesSent[drive.Name] = false;
+            }
+        }
+
+        /// <summary>
+        /// Gets the available RAM space and caclulates if it needs to send an email.
+        /// </summary>
+        /// <param name="monitorItem">The information for the monitor action.</param>
+        /// <param name="threshold">The threshold to check if an email needs to be send.</param>
+        /// <param name="gclCommunicationsService">The name of the service in the configuration, used for logging.</param>
+        /// <param name="configurationServiceName">The communications service from the GCL to actually send out the email.</param>
+        /// <returns></returns>
+        private async Task GetRAMSpaceAsync(ServerMonitorModel monitorItem, int threshold, CommunicationsService gclCommunicationsService, string configurationServiceName)
+        {
+            double ramValue = ramCounter.NextValue();
+            await logService.LogInformation(logger, LogScopes.RunStartAndStop, monitorItem.LogSettings, $"RAM is: {ramValue}MB available", configurationServiceName, monitorItem.TimeId, monitorItem.Order);
+
+            //Set the email settings correctly.
+            receiver = monitorItem.EmailAddressForWarning;
+            subject = "Low on RAM space.";
+            body = $"Your ram has {ramValue}MB available which is below your set threshold of {threshold}MB";
+
+            //Check if the ram is above the threshold.
+            if (ramValue < threshold)
+            {
+                if (!emailRAMSent)
+                {
+                    emailRAMSent = true;
+                    await gclCommunicationsService.SendEmailAsync(receiver, subject, body);
+                }
+            }
+            else
+            {
+                emailRAMSent = false;
+            }
```

Comment:
Wij doen code in zulke gevallen altijd omdraaien, zodat je de code minder ver naar rechts hoeft in te springen, waardoor het beter leesbaar wordt. Dit kan op verschillende plekken in deze PR verbeterd worden, ik geef het alleen nu op deze plek aan als voorbeeld. Dit is een voorbeeld van hoe je het overzichtelijker kunt maken:
```C#
//Check if the ram is above the threshold.
if (ramValue > threshold)
{
	emailRAMSent = false;
	return;
}

if (emailRAMSent)
{
	return;
}

emailRAMSent = true;
await gclCommunicationsService.SendEmailAsync(receiver, subject, body);
```